security-analysis/security-analysis-api/src/test/java/com/powsybl/security/LimitViolationDetectionTest.java

```diff
@@ -78,4 +95,25 @@ protected void checkVoltage(Bus b, int voltageValue, Consumer<LimitViolation> co
     protected void checkVoltageAngle(VoltageAngleLimit voltageAngleLimit, double voltageAngleDifference, Consumer<LimitViolation> consumer) {
         LimitViolationDetection.checkVoltageAngle(voltageAngleLimit, voltageAngleDifference, consumer);
     }
+
+    @Test
+    void testLimitsComputer() {
+        Network network = EurostagTutorialExample1Factory.createWithFixedCurrentLimits();
+        List<LimitReduction> limitReductionList = new ArrayList<>();
+        LimitReduction reduction1 = LimitReduction.builder(LimitType.CURRENT, 0.5)
+            .withMonitoringOnly(false)
+            .withContingencyContext(ContingencyContext.none())
+            .withNetworkElementCriteria(new NetworkElementIdListCriterion(Set.of("NHV1_NHV2_1")))
+            .withLimitDurationCriteria(new PermanentDurationCriterion())
+            .build();
+        limitReductionList.add(reduction1);
+        DefaultLimitReductionsApplier computer = new DefaultLimitReductionsApplier(limitReductionList);
+        checkCurrent(network.getLine("NHV1_NHV2_1"), TwoSides.ONE, 1101, violationsCollector::add, computer);
+        Assertions.assertEquals(1, violationsCollector.size());
+        Assertions.assertEquals(0.5, violationsCollector.get(0).getLimitReduction());
+        Assertions.assertEquals(Integer.MAX_VALUE, violationsCollector.get(0).getAcceptableDuration());
+        Assertions.assertEquals(1101, violationsCollector.get(0).getValue(), 0.01);
```

Comment:
With this value, we cannot see that the reduction is well considered during the violation detection.
It could only be reported in `violationsCollector.get(0).getLimitReduction()`.

To have a full test, you should use a value between 250 and 499.